encrypted-media/scripts/waiting-for-a-key.js

```diff
@@ -0,0 +1,175 @@
+        }).then(function () {
+            // Video should start playing now that it can decrypt the
+            // streams, so wait until a little bit of the video has
+            // played.
+            debugMessage = 'wait_for_timeupdate_event()';
+            return wait_for_timeupdate_event(video);
+        }).catch(function (error) {
+            assert_unreached('Error: '+error.name);
+        });
+
+        // Typical test duration is 6 seconds on release builds
+        // (12 seconds on debug).
+    }, 'Waiting for a key.', {timeout: 20000});
+
+    // Wait for an 'encrypted' event
+    function wait_for_encrypted_event(video)
+    {
+        return new Promise(function (resolve) {
+            video.addEventListener('encrypted', function listener(e) {
+                assert_equals(e.target, video);
+                assert_true(e instanceof window.MediaEncryptedEvent);
+                assert_equals(e.type, 'encrypted');
+                debugEncryptedEventFired = true;
+                video.removeEventListener('encrypted', listener);
+                resolve(e);
+            });
+        });
+    };
+
+    // Wait for a 'waitingforkey' event. Promise resolved when the
+    // event is received.
+    function wait_for_waitingforkey_event(video)
+    {
+        return new Promise(function (resolve) {
+            video.addEventListener('waitingforkey', function listener(e) {
+                assert_equals(e.target, video);
+                assert_equals(e.type, 'waitingforkey');
+                debugWaitingForKeyEventFired = true;
+                video.removeEventListener('waitingforkey', listener);
+                resolve(e);
+            });
+        });
+    };
+
+    // Wait for a 'timeupdate' event. Promise resolved if |video| has
+    // played for more than 0.2 seconds.
+    function wait_for_timeupdate_event(video)
+    {
+        return new Promise(function (resolve) {
+            video.addEventListener('timeupdate', function listener(e) {
+                assert_equals(e.target, video);
+                ++debugTimeUpdateEventCount;
+                if (video.currentTime < 0.2)
+                    return;
+                video.removeEventListener('timeupdate', listener);
+                resolve(e);
+            });
+        });
+    };
+
+    // We need to wait for the message even if for non clearkey DRMs.
+    function wait_for_message_event(mediaKeySession, handler, test)
+    {
+        return new Promise(function (resolve) {
+            mediaKeySession.addEventListener('message', function listener(e) {
+                assert_equals(e.target, mediaKeySession);
+                assert_equals(e.type, 'message');
+                video.removeEventListener('message', listener);
+                return handler(e.messageType, e.message)
+                  .then(function (response) {
+                      return e.target.update(response)
+                  })
+                  .then(function () {
```

Comment:
nit: don't put .then/.catch on the new lines. In all other files here the style is correct.